docs/content/5.providers/directus.md

```diff
@@ -0,0 +1,62 @@
+# Directus
+
+Nuxt Image with Directus integration
+
+Integration between [Directus](https://directus.io/) and the image module.
+
+To use this provider you just need to specify the base URL of your project.
+
+```ts [nuxt.config.ts]
+export default defineNuxtConfig({
+  image: {
+    directus: {
+      // This URL needs to include the final `assets/` directory
+      baseURL: 'http://localhost:8055/assets/',
+    }
+  }
+})
+```
+
+You can easily override default options:
+
+```ts [nuxt.config.ts]
+export default defineNuxtConfig({
+  image: {
+    directus: {
+      baseURL: 'http://mydirectus-domain.com/assets/',
+      modifiers: {
+        withoutEnlargement: 'true',
+        transforms: ['blur:4', 'negate']
+      }
+    }
+  }
+})
+```
+
+## Modifiers
+All the default modifiers from [Directus documentation](https://docs.directus.io/reference/files.html#requesting-a-thumbnail) are available.
+
+```vue
+<nuxt-img
+  provider="directus"
+  src="ad514db1-eb90-4523-8183-46781437e7ee"
+  height="512"
+  fit="inside"
+  quality="20"
+  :modifiers="{ withoutEnlargement: 'true' }"
+/>
+```
+
+Since Directus exposes [the full sharp API](https://sharp.pixelplumbing.com/api-operation) through the `transforms` parameter, we can use it inside our `modifiers` prop:
+
+```vue
+<nuxt-img
+  provider="directus"
+  src="ad514db1-eb90-4523-8183-46781437e7ee"
+  :modifiers="{ height: '512', withoutEnlargement: 'true', transforms: ['blur:4', 'negate'] }"
+/>
+```
+
+::alert{type="info"}
+Directus exposes a `transforms` parameter that accepts a two-dimensional JSON array to communicate directly with the [sharp API](https://sharp.pixelplumbing.com/api-operation). However, for ease of use, we instead decided to expose a single-dimensional array, passing a list of strings as `key:value` or just `key`.
+::
```

Comment:
Heya! Directus maintainer here 👋🏻 

I'm wondering if this should either mirror the Directus type signature exactly to be consistent (so it's easier to learn/ask/find how to use this) _or_ go fully the other way and make this super Vue / JS friendly by supporting it as an object type (like `{ blur: 4 }` instead of `['blur:4']`). I feel like the current solution is a bit of both, but not quite enough of either 😄

Reply:
That sounds good to me. I'd recommend we mirror directus so the docs take less 'translation'

Reply:
Currently I'm a bit split right now (as someone that just got into js/ts):
On one hand, quickly managing a simple list of _secondary_ transforms as a set-and-forget thing is quite appealing. As well as it would be managing a full object for custom scripting.
On the other hand, while the list of lists was easy as an object to deal with for scripting, I've found it quite challenging at first that I had to `[["key",value],["key"]` to do things around. Even more once I got back to this PR and I had forgotten that I was dealing with a list of lists and not with an object.

This is especially true when I'm not using `nuxt-image`. Where maybe I would like to hardcode a transform like `%5B%5B%22blur%22%2C4%5D%2C%5B%22negate%22%5D%5D` into a component to bake a simple effect.

I'm surely biased to go for the simple list or the object, but I do see what could be the "ah ok" confusion between the two docs. ngl I though about going into directus and sharp repos to find out _why_ a list of lists. But currently my js/ts muscles are still too weak. 🙃

Reply:
I would say that this should depend on what type of developer we would like to target.

- **the junior:** the one that likes simple lists.
- **the flex:** the object fan.
- **the pro:** the docs enjoyer.

Reply:
> On the other hand, while the list of lists was easy as an object to deal with for scripting, I've found it quite challenging at first that I had to [["key",value],["key"] to do things around. Even more once I got back to this PR and I had forgotten that I was dealing with a list of lists and not with an object.

The reasoning behind that "double" array is that we don't know nor "care" how many arguments a given function has, so the format is `[function, ...args]`. That in turn means we can expose everything sharp has to offer as a catch-all fallback for the couple of functions we don't make available as the "simple" / friendly way (yet). 

---

Love the analogy, but I don't think it's totally accurate in this case 😄 To me, the "simple" parameters that already exist (eg width/height/format/etc) cover the "junior" and "flex" use case, whereas the transforms should really be seen as a catch-all last-resort for more powerful access to sharp directly. When it comes to that last use case, I'm worried that adding an additional abstraction that only exists in the context of nuxt/image is going to do more harm than good as now it's neither a "nuxt" nor a "directus" thing (which makes it hard to discover/learn) 🙂

Reply:
> The reasoning behind that "double" array is that we don't know nor "care" how many arguments a given function has, so the format is `[function, ...args]`.

True! So that would mean that if I want to use the `normalise` function (that needs two values, a lower and an upper limit) it would become `['normalise', 20,75]` right?

If this is the case I might prefer to revert the array normalization changes and try to simplify the examples in the nuxt-image/docs. @danielroe what do you think?

Reply:
@rijkvanzanten look at this changes [4143b65](https://github.com/nuxt/image/pull/787/commits/4143b652573751eef6661414e8cb49fcc20dc541):

Basically I'm directly managing the transform as a list of lists and encoding it in urlsafe automagically.


p.s.: I'm noticing that I don't know how to reference line changes in a comment or suggest a review for a specific commit.

Reply:
> If this is the case I might prefer to revert the array normalization changes and try to simplify the examples in the nuxt-image/docs.

Perfect 👍

Reply:
Ok, marking this one as resolved too. We are back to a list of lists and made sure to encode it properly.

Reply:
> Hehe getting a bit off-topic in this PR, but do make sure to upvote and leave your thoughts on https://github.com/directus/directus/discussions/13041 🙂

This PR made me think of a nice usecase, I'll make sure to write it in that discussion (that I was still reading thoroughly).

> You can (shift-)click the line-numbers in the file to update the URL in the files tab to be a permalink to those lines 🙂
>
> https://github.com/nuxt/image/pull/787/files#diff-5ce8ceb4b0a441713600d54be499a9d005b00077aa2ef59009d492eb6fa422d7R43-R44

See? It graphically didn't render, but maybe it's a github thing. I'll investigate in a different moment.